api/core/workflow/nodes/base/node.py

```diff
@@ -269,6 +304,32 @@ def version(cls) -> str:
         # in `api/core/workflow/nodes/__init__.py`.
         raise NotImplementedError("subclasses of BaseNode must implement `version` method.")
 
+    @classmethod
+    def get_node_type_classes_mapping(cls) -> Mapping["NodeType", Mapping[str, type["Node"]]]:
+        """Return mapping of NodeType -> {version -> Node subclass} using __init_subclass__ registry.
+
+        We still best-effort import all modules under core.workflow.nodes so subclasses register
+        themselves on import. Then we return a shallow copy of the registry to avoid accidental mutation.
+        """
+        # Best-effort import to ensure all node modules are loaded (thus registered)
+        try:
```

Comment:
I'm wondering if this is worth the cost. If we support a custom node registry, we would need to manually import node classes to ensure they are properly registered, especially since they might not reside inside the `core.workflow.nodes` package.

That said, I'm fine with keeping the current auto-discovery logic.